lib/logic/player/favorites.dart

```diff
@@ -54,7 +54,7 @@ class FavoritesControl with Control {
     final Set<int> favoriteSet = {};
     if (_useMediaStoreFavorites(contentType)) {
       for (final song in ContentControl.instance.state.allSongs.songs) {
-        if (song.isFavoriteInMediaStore!) {
```

Comment:
This code must not be called on API <30

Currently it's filtered out by [this condition](https://github.com/nt4f04uNd/sweyer/blob/master/lib/logic/player/favorites.dart#L27), that's why there's a non-nullable assertion `!` there

If it would be called, I would like for this code to rather crash, than silently be executed

Reply:
`isFavoriteInMediaStore` is just an attribute that we get from the Android media database and we've seen that they can be null, also on API >=30. I really don't think we should allow our app to crash just because that value can be null, especially here during initialization which will prevent the whole app from loading up.

An alternative approach would be to change our model and make it not-nullable, and fall back to `false` if it is null in the media database.

Reply:
I'd prefer for it to have a default value then, yeah

Reply:
Actually, I see that on the Kotlin side on R and above we send `getInt() == 1`, so it will never be null in Dart

So technically, it can be null, but practically it currently will not

Still, putting a default value would be OK

Reply:
> Actually, I see that on the Kotlin side on R and above we send getInt() == 1, so it will never be null in Dart

Ah yes, indeed. But I agree, let's use a default value and make it not nullable.